airflow/providers/google/cloud/operators/dataform.py

```diff
@@ -0,0 +1,409 @@
+        )
+        workflow_invocation_id = result.name.split("/")[-1]
+        DataformWorkflowInvocationLink.persist(
+            operator_instance=self,
+            context=context,
+            project_id=self.project_id,
+            region=self.region,
+            repository_id=self.repository_id,
+            workflow_invocation_id=workflow_invocation_id,
+        )
+        if not self.asynchronous:
+            hook.wait_for_workflow_invocation(
+                workflow_invocation_id=workflow_invocation_id,
+                repository_id=self.repository_id,
+                project_id=self.project_id,
+                region=self.region,
+                timeout=self.timeout,
+            )
+        return WorkflowInvocation.to_dict(result)
+
+
+class DataformGetWorkflowInvocationOperator(BaseOperator):
+    """
+    Fetches a single WorkflowInvocation.
+
+    :param project_id: Required. The ID of the Google Cloud project that the task belongs to.
+    :param region: Required. The ID of the Google Cloud region that the task belongs to.
+    :param repository_id: Required. The ID of the Dataform repository that the task belongs to.
+    :param workflow_invocation_id:  the workflow invocation resource's id.
+    :param retry: Designation of what errors, if any, should be retried.
+    :param timeout: The timeout for this request.
+    :param metadata: Strings which should be sent along with the request as metadata.
+    :param gcp_conn_id: The connection ID to use when fetching connection info.
+    :param delegate_to: The account to impersonate, if any. For this to work, the service accountmaking the
+        request must have  domain-wide delegation enabled.
+    :param impersonation_chain: Optional service account to impersonate using short-term
+        credentials, or chained list of accounts required to get the access_token
+        of the last account in the list, which will be impersonated in the request.
+        If set as a string, the account must grant the originating account
+        the Service Account Token Creator IAM role.
+        If set as a sequence, the identities from the list must grant
+        Service Account Token Creator IAM role to the directly preceding identity, with first
+        account from the list granting this role to the originating account (templated).
+    """
+
+    template_fields = ("repository_id", "workflow_invocation_id", "delegate_to", "impersonation_chain")
+    operator_extra_links = (DataformWorkflowInvocationLink(),)
+
+    def __init__(
+        self,
+        project_id: str,
+        region: str,
+        repository_id: str,
+        workflow_invocation_id: str,
+        retry: Union[Retry, _MethodDefault] = DEFAULT,
+        timeout: Optional[float] = None,
+        metadata: Sequence[Tuple[str, str]] = (),
+        gcp_conn_id: str = "google_cloud_default",
+        delegate_to: Optional[str] = None,
+        impersonation_chain: Optional[Union[str, Sequence[str]]] = None,
```

Comment:
Thinking aloud, outside of this PR. Could we replace impersonation_chain ugly type hint with something nicer?

From
```
Optional[Union[str, Sequence[str]]]
```
to
```
ImpersonationChain
```
(not sure what is possible with python type hints)

Reply:
This is something that can be improved in other PR, because we should change in all google operators at one time.

Reply:
Yeah, that's why I mentioned that it's outside of this PR. I will investigate more into typing in meantime and see if we can improve it in the future.